api/user.js

```diff
@@ -0,0 +1,46 @@
+import { BAD_REQUEST, TYPE_JSON } from './defs.js'
+import { POST } from './router.js'
+import * as db from './db.js'
+
+// function to check the form's data
+const verifyInformations = (body) => {
+   let checks = [];
+   for(const [name,item] of Object.entries(body)){
+      switch(name){
+        case "own_contact":
+           isNaN(item) && checks.push('please enter your phone number')
+           break;
+        case "emergency_contact":
+          isNaN(item) && checks.push("please enter a phone number to join in case of emergency")
+          break;
+        default:
+          break;
+     }
+   }
+   return checks
+};
+
+POST.user.register_form = async ({session,request}) => {
```

Comment:
I don't think we want to have `_form` in our route, and the user is already register, it should be a simple `POST` on the user I think.

The user need to be authenticated thought to be able to modify his session, you can use `withUser` to ensure the user exists